### PR TITLE
add devices and channels validation schema + some tests

### PR DESCRIPTION
**Pull request by manuIO**

### CODE REVIEW

File: api/channels.go [hunk 1]
```diff
@@ -39,30 +39,43 @@ type (
...
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	c := models.Channel{}
-	if len(body) > 0 {
-		if err := json.Unmarshal(body, &c); err != nil {
+	if len(data) > 0 {
+		err, str := validateChannelSchema(data)
```
Comment (mijicd) [inline api/channels.go]: It's possible to inline these two lines, as you did in line 60.

File: api/channels.go [hunk 2]
```diff
@@ -163,27 +176,40 @@ func updateChannel(w http.ResponseWriter, r *http.Request) {
...
-	c := models.Channel{}
-	if len(body) > 0 {
-		if err := json.Unmarshal(body, &c); err != nil {
+	// Validate JSON schema
+	if len(data) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		str := `{"response": "no data provided"}`
+		io.WriteString(w, str)
+		return
+	} else {
```
Comment (mijicd) [inline api/channels.go]: Avoid else branches where possible.

File: api/validate_schema.go [hunk 3]
```diff
@@ -69,7 +69,13 @@ func validateDeviceSchema(data []byte) (bool, string) {
 				return true, str
 			case "name":
 				if (len(body[k].(string)) > 20) {
-					str := `{"response": "max name size: 20"}`
+					str := `{"response": "max name size 20"}`
+					return true, str
+				}
+				break
+			case "description":
+				if (len(body[k].(string)) > 200) {
```
Comment (manuIO) [inline api/validate_schema.go]: fixed here: https://github.com/mainflux/mainflux-core/pull/12/commits/b5b7f6f69b11f7d3d3ee8f683fea743aa3498981

File: api/channels.go (hunk 2 above)
Comment (manuIO) [inline api/channels.go]: fixed here: https://github.com/mainflux/mainflux-core/pull/12/commits/8ff3d8612c28e639811e46d7aa4fbf53f721c258

**Comment by drasko** [comment]
:clap:

**Pull request merged**